### PR TITLE
Holdings administrative notes

### PR DESCRIPTION
**Pull request by natalyapik**
new metadb derived table for holdings_administrative_notes

### CODE REVIEW

**Review by jenmawe — CHANGES_REQUESTED** [review]
All queries:
- [x] Pull request has a complete description
- [x] Query runs without errors
- [x] Query output is correct
- [x] Query logic is clear and well documented
- [x] Query is readable and properly indented
- [x] Table and column names are in all-lowercase
- [x] Quotation marks are used only where necessary

Derived tables:
- [x] Query begins with user documentation in comment lines
**- [] File name is listed in `runlist.txt` after dependencies**
- [x] All columns have indexes
- [x] Table is vacuumed and analyzed

**Comment by nassibnassar** [comment]
- [ ] Pull request has a complete description
- [ ] Query runs without errors
- [ ] Query output is correct
- [ ] Query logic is clear and well documented
- [ ] Query is readable and properly indented
- [ ] Table and column names are in all-lowercase
- [ ] Quotation marks are used only where necessary
- [ ] Query begins with user documentation in comment lines
- [ ] File name is listed in `runlist.txt` after dependencies
- [ ] All columns have indexes
- [ ] Table is vacuumed and analyzed

File: sql_metadb/derived_tables/holdings_administrative_notes.sql [hunk 1]
```diff
@@ -0,0 +1,26 @@
+-- This derived table extracts administrative notes from the holdings record.
+DROP TABLE IF EXISTS holdings_administrative_notes;
+
+CREATE TABLE holdings_administrative_notes AS 
+SELECT 
+    hd.instance_id,
+    hd.id AS holdings_id,
+    hd.hrid AS holdings_hrid,
+    adminNotes.jsonb #>> '{}' AS administrative_notes,
```
Comment (lib-eric) [inline sql_metadb/derived_tables/holdings_administrative_notes.sql:9]: Advised to use singular column name and all-lowercase table names
```suggestion
    admin_notes.jsonb #>> '{}' AS administrative_note,
```

File: sql_metadb/derived_tables/holdings_administrative_notes.sql [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    adminNotes.ordinality AS administrative_notes_ordinality
```
Comment (lib-eric) [inline sql_metadb/derived_tables/holdings_administrative_notes.sql:10]: Advised to use singular column name and all-lowercase table names
```suggestion
    admin_notes.ordinality AS administrative_note_ordinality
```

File: sql_metadb/derived_tables/holdings_administrative_notes.sql [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+FROM 
+    folio_inventory.holdings_record__t AS hd
+    LEFT JOIN folio_inventory.holdings_record AS hld ON hd.id = hld.id 
+    CROSS JOIN LATERAL jsonb_array_elements(jsonb_extract_path(hld.jsonb, 'administrativeNotes')) WITH ORDINALITY AS adminNotes (jsonb);
```
Comment (lib-eric) [inline sql_metadb/derived_tables/holdings_administrative_notes.sql:14]: All-lowercase table names
```suggestion
    CROSS JOIN LATERAL jsonb_array_elements(jsonb_extract_path(hld.jsonb, 'administrativeNotes')) WITH ORDINALITY AS admin_notes (jsonb);
```

File: sql_metadb/derived_tables/holdings_administrative_notes.sql [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    
+CREATE INDEX ON holdings_administrative_notes (instance_id);
+
+CREATE INDEX ON holdings_administrative_notes (holdings_id);
+
+CREATE INDEX ON holdings_administrative_notes (holdings_hrid);
+
+CREATE INDEX ON holdings_administrative_notes (administrative_notes);
```
Comment (lib-eric) [inline sql_metadb/derived_tables/holdings_administrative_notes.sql:22]: Advised to use singular column name
```suggestion
CREATE INDEX ON holdings_administrative_notes (administrative_note);
```

File: sql_metadb/derived_tables/holdings_administrative_notes.sql [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+CREATE INDEX ON holdings_administrative_notes (administrative_notes_ordinality);
```
Comment (lib-eric) [inline sql_metadb/derived_tables/holdings_administrative_notes.sql:24]: Advised to use singular column name
```suggestion
CREATE INDEX ON holdings_administrative_notes (administrative_note_ordinality);
```

**Review by lib-eric — CHANGES_REQUESTED** [review]
Changes for singular column names

All queries:
- [x] Pull request has a complete description
- [x] Query runs without errors
- [x] Query output is correct
- [x] Query logic is clear and well documented
- [x] Query is readable and properly indented
- [ ] Table and column names are in all-lowercase
- [x] Quotation marks are used only where necessary

Derived tables:
- [x] Query begins with user documentation in comment lines
- [x] File name is listed in `runlist.txt` after dependencies
- [x] All columns have indexes
- [x] Table is vacuumed and analyzed

**Review by lib-eric — APPROVED** [review]
Looks good - thank you!

**Review by jenmawe — CHANGES_REQUESTED** [review]
We use holdings as in a holdings record. Could you update holding_id to holdings_id and the others like this?

File: sql_metadb/derived_tables/holdings_administrative_notes.sql [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+FROM 
+    folio_inventory.holdings_record__t AS hd
+    LEFT JOIN folio_inventory.holdings_record AS hld ON hd.id = hld.id 
+    CROSS JOIN LATERAL jsonb_array_elements(jsonb_extract_path(hld.jsonb, 'administrativeNotes')) WITH ORDINALITY AS admin_notes (jsonb);
+    
+CREATE INDEX ON holdings_administrative_notes (instance_id);
+
+CREATE INDEX ON holdings_administrative_notes (holdings_id);
+
+CREATE INDEX ON holdings_administrative_notes (holdinsg_hrid);
```
Comment (nassibnassar) [inline sql_metadb/derived_tables/holdings_administrative_notes.sql:20]: Typo

**Review by jenmawe — APPROVED** [review]

**Pull request merged**